portal/server/helpers/filterReportJson.js

```diff
@@ -0,0 +1,23 @@
+import { UTILISATION_REPORT_HEADERS } from '@ukef/dtfs2-common';
+
+/**
+ * Iterates through an array of objects and removes any key value pairs
+ * from each object where the key isn't in the UTILISATION_REPORT_HEADERS const
+ * @param {Array<Object>} reportJson auth token
+ * @returns {Object} Filtered down object
+ */
+export const filterReportJsonToRelevantKeys = (reportJson) => {
```

Comment:
`map` could be slower when dealing with large number of rows I would suggest something like below to improve execution efficiency:

```javascript
export const filterReportJsonToRelevantKeys = (reportJson) => {
  // Convert relevant keys into a Set for O(1) lookups
  const relevantKeysSet = new Set(Object.values(UTILISATION_REPORT_HEADERS));

  // Map through the report rows and filter keys
  return reportJson.map((reportRow) => {
    return Object.fromEntries(
      Object.entries(reportRow).filter(([key]) => relevantKeysSet.has(key))
    );
  });
};
```

Reply:
For reference I ran the script with 5 iterations across both functions (original and proposed) with 1000 rows in the reportJson and 15 columns getting filtered down to 5 columns, I can do 31 -> 11 which is our real world example to double check